warden/warden_daemon/tests/common/mod.rs

```diff
@@ -0,0 +1,100 @@
+use std::{
+    env,
+    fs::{remove_dir_all, remove_file},
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+use nix::{
+    sys::signal::{self, Signal::SIGINT},
+    unistd::Pid,
+};
+use tokio_vsock::VMADDR_CID_HOST;
+use uuid::Uuid;
+use warden_client::realm::RealmConfig;
+use warden_client::realm::{CpuConfig, KernelConfig, MemoryConfig, NetworkConfig};
+use warden_daemon::cli::Cli;
+
+pub struct PathResourceManager {
+    resource_path: Box<dyn AsRef<Path>>,
+}
+
+impl PathResourceManager {
+    pub async fn new() -> Self {
+        const TEST_FOLDER_PATH: &str = "/tmp/warden-daemon-integration-tests";
+        tokio::fs::create_dir_all(TEST_FOLDER_PATH).await.unwrap();
+        Self {
+            resource_path: Box::new(format!("{}/{}", TEST_FOLDER_PATH, Uuid::new_v4())),
+        }
+    }
+
+    pub fn get_path(&self) -> &Path {
+        (*self.resource_path).as_ref()
+    }
+}
+
+impl Drop for PathResourceManager {
+    fn drop(&mut self) {
+        let path = self.get_path();
+        if let Err(_) = remove_dir_all(path) {
```

Comment:
Was this failing? From what I read the remove_dir_all removes the directory with everything that is inside. Why did you need to add this if statement? From what I understand the path will always be a path to a directory as it is created this way in the new() function.